### PR TITLE
Rooms front end test

### PR DESCRIPTION
**Pull request by hadicheaito1**
## Description

- Front end test for Rooms page, using Jest
## Testing Steps

- Pull code
- run ```jest -u```
- Ensure all tests pass
## Screenshots (if appropriate):

image

### CODE REVIEW

**Comment by scouillard** [comment]
It seems like every aria-label, aria-control and id are dynamically rendered on every test run, which make the snapshot test fail every time.

I think one way to solve the issue would be to expect any() for the listed properties above.
https://jestjs.io/docs/snapshot-testing#property-matchers

Or to hard code the properties.
https://stackoverflow.com/questions/53354341/react-jest-snapshot-testing-with-dynamic-id

**Comment by scouillard** [comment]
Running `jest -u` will re-generate snapshot artifacts for all failing snapshot tests, and should be used only when there is change in the DOM/layout.

File: app/javascript/__tests__/rooms/Rooms.test.jsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      {
+        name: 'Room 2',
+        friendly_id: 'dummy_id3',
+      },
+    ],
+};
+
+describe('Rooms (useRooms query test)', () => {
+  test('Rooms from query match expected rooms', () => {
+    const json = useRooms();
```
Comment (KH-Amir-TN) [inline app/javascript/__tests__/rooms/Rooms.test.jsx:31]: We are testing a mock here. 
To test queries we should mock the API calls but use the queries original implementation.

File: app/javascript/__tests__/rooms/Rooms.test.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import renderer from 'react-test-renderer';
+import Rooms from '../../components/rooms/Rooms';
+import RoomsList from '../../components/rooms/RoomsList';
+import RoomsTabs from '../../components/rooms/RoomsTabs';
+import useRooms from '../../hooks/queries/rooms/__mocks__/useRooms';
+
+jest.mock('react-router-dom');
+jest.mock('../../hooks/queries/rooms/useRooms');
+
+const rooms = {
```
Comment (KH-Amir-TN) [inline app/javascript/__tests__/rooms/Rooms.test.jsx:11]: I think this is not useful since it's in place only to test a mock.

**Review by KH-Amir-TN — CHANGES_REQUESTED** [review]

**Comment by KH-Amir-TN** [comment]
> It seems like every aria-label, aria-control and id are dynamically rendered on every test run, which make the snapshot test fail every time.
> 
> I think one way to solve the issue would be to expect any() for the listed properties above via Property Matchers. https://jestjs.io/docs/snapshot-testing#property-matchers
> 
> Or to hard code the properties directly in the prop. https://stackoverflow.com/questions/53354341/react-jest-snapshot-testing-with-dynamic-id
> 
> I've been trying both solutions so far without success.

You had this problem when pulling this PR or on one that you're working on?

**Comment by scouillard** [comment]
> > It seems like every aria-label, aria-control and id are dynamically rendered on every test run, which make the snapshot test fail every time.
> > I think one way to solve the issue would be to expect any() for the listed properties above via Property Matchers. https://jestjs.io/docs/snapshot-testing#property-matchers
> > Or to hard code the properties directly in the prop. https://stackoverflow.com/questions/53354341/react-jest-snapshot-testing-with-dynamic-id
> > I've been trying both solutions so far without success.
> 
> You had this problem when pulling this PR or on one that you're working on?

On this PR.

**Pull request closed without merging**